File: java/ZH1_1/elso/src/Playlist.java
import java.util.ArrayList;
import java.util.List;

public class Playlist {
    private static List<Game> games;

    public Playlist() {
        games = new ArrayList<>();
    }

    public void add(Game name){
        games.add(name);
    }

    public Game getBestGame(){
        Game game = null;
        int max = 0;
        for (Game g:games) {
            if (g.getScore() > max) {
                max = g.getScore();
                game = g;
            }
        }

        return game;
    }

    public Game getLongestGame(){
        Game game = null;
        int max = 0;
        for (Game g:games) {
            if (g.getTime() > max) {
                max = g.getScore();
                game = g;
            }
        }

        return game;
    }
    public Game getShortestGame(){
        Game game = null;

        double max = 0;
        for (Game g:games) {
            if (g.getTime() > max) {
                max = g.getScore();
            }
        }
        double min = max;
        for (Game g:games) {
            if (g.getTime() < min) {
                min = g.getTime();
                game = g;
            }
        }

        return game;
    }


}

File: cs/ZH1_1/feladat1/ZH1_1/Playlist.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ZH1_1
{
    internal class Playlist
    {
        private List<Game> playlist;

        public Playlist()
        {
            this.playlist = new List<Game>();
        }

        public void add(Game game)
        {
            playlist.Add(game);
        }
        public Game getBestGame() 
        {
            Game bestGame = null;
            int max = 0;
            foreach (Game game in playlist)
            {
                if (game.getScore() > max) {
                    max = game.getScore();
                    bestGame = game;
                }
            }

            return bestGame;
        }
        
        public Game getLongestGame() 
        {
            Game logestGame = null;
            double max = 0;
            foreach (Game game in playlist)
            {
                if (game.getTime() > max) {
                    max = game.getTime();
                    logestGame = game;
                }
            }

            return logestGame;
        }
        public Game getShortestGame() 
        {
            Game shortestGame = null;
            double min = getLongestGame().getTime();
            foreach (Game game in playlist)
            {
                if (game.getTime() < min) {
                    min = game.getTime();
                    shortestGame = game;
                }
            }

            return shortestGame;
        }
    }
}
